### DISCUSSION TITLE
Horizontal scroll implementation

### DISCUSSION THREAD

**Discussion by hiqua** [iSoron/uhabits#1060] [Technical Discussion]
@NomarCub let's continue the discussion here as suggested.

About https://github.com/iSoron/uhabits/discussions/466#discussioncomment-1154180: I'd say anything that just works is a good start, whether that will lead you to something close to be considered for merging is another story. If you use this trick, do the problems suggested in https://github.com/iSoron/uhabits/discussions/466#discussioncomment-1127907 manifest themselves? Or does everything works fine?

**Comment by NomarCub** [comment]
I see no specific issues in the discussion you linked. It worked fine for the short time I tried it, so I guess no issues.

**Reply by hiqua** [reply]
The reminders work as expected?

**Reply by NomarCub** [reply]
I haven't tried yet. I'll do it when I have the time. I haven't used reminders that much.

Working as expected, meaning they are not offset by hours or days? How are not everyday habit reminders supposed to work? Notify you only when you won't have done it x out of y days by today (no hollow checkmark for today)?

Is there an easy way to change the package name for testing? I wouldn't like it to affect my regular workflow.

**Reply by hiqua** [reply]
> Is there an easy way to change the package name for testing? I wouldn't like it to affect my regular workflow.

An even easier way would be to transfer the .apk, use Shelter (https://f-droid.org/en/packages/net.typeblog.shelter/) to enable the work profile, and install your version there.

But this seems to work as well (just tried it):
* change `applicationId` to something else
* change `android:authorities` in AndroidManifest.xml to something else, e.g. the same value as before

**Reply by NomarCub** [reply]
Thanks!
Changing package names in those two places makes it possible to install, but the app crashes on notifications, even without the experimental thing enabled.
Shelter works fine.
It would've been nice to separate the experimental bit from my regular habit tracking, even down to app versions, but this'll do.
I'll report back with the results.

**Reply by hiqua** [reply]
Actually with Shelter you'll still have only one version of the app installed with different data, so double-check that you actually have a state you're comfortable with now, and still make a backup before experimenting more.

**Reply by NomarCub** [reply]
Yeah, that's what I meant, that I would've liked to have two different app versions, but I can't do it properly with either method. The app's data export is satisfactory, and I don't use the past midnight feature anyway, so it's fine this way too.
\
\
I checked out what happens to notifications. They function as if it was the day of the offset. It doesn't shift hours if you set something not a multiple of a 24 hours, but you get it for the wrong day still.

I set [this](https://github.com/iSoron/uhabits/blob/dev/uhabits-android/src/main/java/org/isoron/uhabits/HabitsApplication.kt#L74) from 3 to -7*24 locally, then toggle the `Extend day a few hours past midnight option`. This gives me kind of a weekly planning mode that I can switch to occasionally, that I'm nearly satisfied with already.

**Reply by hiqua** [reply]
Ok, feel free to open a PR as soon as you have something that could be acceptable.

**Reply by NomarCub** [reply]
Thanks for the help! I will, but probably not soon at all. I'd rather encourage others to give this a shot.

I'm not really clued into the workings of the app, and not keen on digging in more, given how many other things you said this might affect.

**Reply by hiqua** [reply]
> I'd rather encourage others to give this a shot.

I think if you want this feature to land you're actually your best hope right now, but up to you!
